From e029ea6881a24cffa38300fafe4593bdacbdca87 Mon Sep 17 00:00:00 2001
From: Chongyang Xu <cxu@mpi-sws.org>
Date: Thu, 18 Apr 2024 13:21:40 +0000
Subject: [PATCH] third_party:tensorpipe:third_party:libuv: comparing signed
 and unsigned

---
 src/unix/thread.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/unix/thread.c b/src/unix/thread.c
index 1a85d1d4..e3994561 100644
--- a/src/unix/thread.c
+++ b/src/unix/thread.c
@@ -192,7 +192,7 @@ static size_t thread_stack_size(void) {
      * on the architecture.
      */
     if (lim.rlim_cur >= 8192)
-      if (lim.rlim_cur >= PTHREAD_STACK_MIN)
+      if (lim.rlim_cur >= (long unsigned int)PTHREAD_STACK_MIN)
         return lim.rlim_cur;
   }
 #endif
@@ -240,8 +240,8 @@ int uv_thread_create_ex(uv_thread_t* tid,
     /* Round up to the nearest page boundary. */
     stack_size = (stack_size + pagesize - 1) &~ (pagesize - 1);
 #ifdef PTHREAD_STACK_MIN
-    if (stack_size < PTHREAD_STACK_MIN)
-      stack_size = PTHREAD_STACK_MIN;
+    if (stack_size < (long unsigned int)PTHREAD_STACK_MIN)
+      stack_size = (long unsigned int)PTHREAD_STACK_MIN;
 #endif
   }
 
-- 
2.34.1

